### PR TITLE
Add nested recipe check to get_recipes

### PR DESCRIPTION
**Pull request by mikecormier**
Currently get_recipes only support recipes nested up to two deep from the recipe_folder provided. Our application of get_recipes, and potential future uses from others, has recipes nested deeper. This purposed implementation will provide the ability to locate all recipes, no matter how deep, within the user provided recipe_folder.

### CODE REVIEW

**Comment by daler** [comment]
Thanks! Would you please add a test for this in `test/test_utils.py`?

**Comment by mikecormier** [comment]
Yes, I can add a test. 
It looks like 1 of the 3 test in ci/circleci: test-linux (long_running) failed, however, I don't believe this is due to the changes I made. Can you confirm this?

**Comment by daler** [comment]
Thanks. Not sure yet what's going on in the failed test, but it's happening in the cb3-migration branch so it's not due to your changes.

**Comment by mikecormier** [comment]
Hey @daler,
I have added a test for the updated get_recipes. The `test-linux (long_running)` fails at the same place, but everything else is passing. These changes are ready to be merged. Could you merge them into the cb3-migration branch?

**Comment by johanneskoester** [comment]
@mikecormier could you merge in the latest version of cb3-migration branch?

**Comment by mikecormier** [comment]
@johanneskoester I have merged the latest updates in the cb3-migration into my branch and all test are passing. This PR is ready to be merge.

File: bioconda_utils/utils.py [hunk 1]
```diff
@@ -531,10 +531,10 @@ def get_recipes(recipe_folder, package="*"):
         logger.debug(
             "get_recipes(%s, package='%s'): %s", recipe_folder, package, p)
         path = os.path.join(recipe_folder, p)
-        yield from map(os.path.dirname,
-                       glob.glob(os.path.join(path, "meta.yaml")))
-        yield from map(os.path.dirname,
-                       glob.glob(os.path.join(path, "*", "meta.yaml")))
+        for newDir in [x for x in glob.glob(path)]:
```
Comment (johanneskoester) [inline bioconda_utils/utils.py]: Please use snake_case for variable names.

Comment (johanneskoester) [reply bioconda_utils/utils.py]: This `[x for x in glob.glob(path)]` can simply be written as `glob.glob(path)`.

File: test/test_utils.py [hunk 2]
```diff
@@ -976,3 +976,98 @@ def test_compiler():
         for i in utils.built_package_paths(v):
             assert os.path.exists(i)
             ensure_missing(i)
+              
+            
+def test_depthcheck_get_recipe():
```
Comment (johanneskoester) [inline test/test_utils.py]: Rather name this `test_nested_recipes`?

**Comment by johanneskoester** [comment]
cb-3migration is merged into master now. Please switch to master as merge target.

**Comment by mikecormier** [comment]
@johanneskoester, 
I have made the changes you requested and all test are passing.

**Comment by johanneskoester** [comment]
Looks good to me!

**Pull request merged**